Infrastructure/checkService.js

```diff
@@ -0,0 +1,70 @@
+const ping = require('ping');
+const exec = require('child_process').exec;
+const dig = require('node-dig-dns');
+const {
+  servers: serverList,
+  workstations: workstationList,
+  dnsQueryDomain,
+  sshUser
+} = require('./config.js');
+
+const DIG_ARG = 'ns';
+const SSH_TIMEOUT = 3;
+
+const pingWorkstation = workstation => {
+  return new Promise(resolve => {
+    ping.sys.probe(workstation.address, isAlive => {
```

Comment:
ping seems to have [a promise-based option](https://www.npmjs.com/package/ping#promise-wrapper) as well, though that may not be as useful since the result needs to be mapped anyways.

Reply:
I think it looks cleaner promise-based. Added the changes.